src/features/UserForm/ui/UncontrolledForm.tsx

```diff
@@ -0,0 +1,289 @@
+
+import { addEntry } from '../model/formsSlice';
+
+import {
+  AppButton,
+  FormCheckbox,
+  FormInput,
+  RadioGroup,
+  AutocompleteInput,
+  FileInput,
+  PasswordInput,
+} from '@/shared/ui';
+import { passwordChecks, fileToBase64 } from '@/shared/utils';
+
+type UncontrolledFormProps = { onSuccess?: () => void };
+
+export const UncontrolledForm = ({ onSuccess }: UncontrolledFormProps) => {
+  const dispatch = useDispatch();
+
+  const [nameError, setNameError] = useState<string | null>(null);
+  const [ageError, setAgeError] = useState<string | null>(null);
+  const [emailError, setEmailError] = useState<string | null>(null);
+  const [genderError, setGenderError] = useState<string | null>(null);
+  const [countryError, setCountryError] = useState<string | null>(null);
+  const [passwordError, setPasswordError] = useState<string | null>(null);
+  const [confirmPasswordError, setConfirmPasswordError] = useState<
+    string | null
+  >(null);
+  const [pictureError, setPictureError] = useState<string | null>(null);
+  const [termsError, setTermsError] = useState<string | null>(null);
+
+  const nameRef = useRef<HTMLInputElement | null>(null);
+  const ageRef = useRef<HTMLInputElement | null>(null);
+  const emailRef = useRef<HTMLInputElement | null>(null);
+  const genderFirstRef = useRef<HTMLInputElement | null>(null);
+  const countryInputRef = useRef<HTMLInputElement | null>(null);
+  const passwordRef = useRef<HTMLInputElement | null>(null);
+  const confirmPasswordRef = useRef<HTMLInputElement | null>(null);
+  const pictureRef = useRef<HTMLInputElement | null>(null);
+  const termsRef = useRef<HTMLInputElement | null>(null);
+
+  const [country, setCountry] = useState('');
+
+  const [strength, setStrength] = useState({
+    number: false,
+    upper: false,
+    lower: false,
+    special: false,
+  });
+
+  const handlePasswordChange = (val: string) => {
+    setStrength({
+      number: passwordChecks.number.test(val),
+      upper: passwordChecks.upper.test(val),
+      lower: passwordChecks.lower.test(val),
+      special: passwordChecks.special.test(val),
+    });
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const formEl = e.currentTarget;
+
+    const formData = new FormData(formEl);
+    const name = formData.get('name')?.toString().trim() ?? '';
+    const ageRaw = formData.get('age')?.toString().trim() ?? '';
+    const email = formData.get('email')?.toString().trim() ?? '';
+    const password = formData.get('password')?.toString() ?? '';
+    const confirmPassword = formData.get('confirmPassword')?.toString() ?? '';
+    const gender = (formData.get('gender')?.toString() ?? 'other') as
+      | 'female'
+      | 'male'
+      | 'other';
+    const accepted = formData.get('acceptTerms') === 'on';
+    const file = pictureRef.current?.files?.[0];
+    const age = Number(ageRaw);
+
+    let hasError = false;
+
+    if (!name) {
```

Comment:
in every condition we do the same things: set error, focus field and set errors in true.

I think it's doable to make one function and reuse it to have less code.

like this: const validateField = (condition:boolean, setError (message: string)=>void, message: stirng, ref: RefObject, hasErrors: boolean) =>{ 
     if(condition){
        setError(message);
        if(!hasErrors){
            ref.current.focus();
            return true;
        }
        return false
     }
}

something like this